index.js

```diff
@@ -1,394 +1,330 @@
-    constructPayload(rudderPayload, pHEvent, constants, true);
+    constructPayload(rudderPayload, event, constants, true)
 
     // add generic props
-    constructPayload(rudderPayload, pHEvent, generic);
+    constructPayload(rudderPayload, event, generic)
 
     // get specific event props
-    let eventName = get(pHEvent, "event");
-    let { type, mapping } = eventToMapping[eventName]
-      ? eventToMapping[eventName]
-      : eventToMapping["default"];
+    const eventName = get(event, 'event')
+    const { type, mapping } = eventToMapping[eventName] ? eventToMapping[eventName] : eventToMapping['default']
 
     //set Rudder payload type
-    set(rudderPayload, "type", type);
+    set(rudderPayload, 'type', type)
+
     // set Rudder event props
-    constructPayload(rudderPayload, pHEvent, mapping);
+    constructPayload(rudderPayload, event, mapping)
 
     // add all pther posthog keys under props not starting with $ to Rudder payload properties
-    Object.keys(pHEvent.properties).forEach(propKey => {
-      if (
-        propKey.slice(0, 1) != "$" &&
-        pHEvent.properties[propKey] != undefined &&
-        pHEvent.properties[propKey] != null
-      ) {
-        set(
-          rudderPayload,
-          `properties.${propKey}`,
-          pHEvent.properties[propKey]
-        );
-      }
-    });
-
-    rudderEventsArray.push(rudderPayload);
-  });
-  batch.batch = rudderEventsArray;
-  batch.sentAt = new Date().toISOString();
-  await sendToRudder(global, cache, batch);
-  return events;
-}
+    Object.keys(event.properties).forEach((propKey) => {
+        if (propKey.slice(0, 1) != '$' && event.properties[propKey] != undefined && event.properties[propKey] != null) {
+            set(rudderPayload, `properties.${propKey}`, event.properties[propKey])
+        }
+    })
 
-async function sendToRudder(global, cache, batch) {
-  let olderBatch, payload;
-  try {
-    olderBatch = await cache.get(global.writeKey);
-    olderBatch = JSON.parse(olderBatch);
-    console.log("oldbatch: " + JSON.stringify(olderBatch));
-    payload = {
-      batch: olderBatch
-        ? [...olderBatch.batch, ...batch.batch]
-        : [...batch.batch],
-      sentAt: new Date().toISOString()
-    };
-    let res = await fetchWithRetry(
-      global.dataPlaneUrl,
-      {
-        headers: {
-          "Content-Type": "application/json",
-          ...global.rudderAuthHeader.headers
-        },
-        body: payload
-      },
-      "POST"
-    );
-    console.log(
-      "***sending payload to Rudder server gave a response, deleting any cached data******" +
-        res.status
-    );
-    await cache.expire(global.writeKey, 0);
-  } catch (err) {
-    console.log("fetchWithRetry thrown error: " + err);
-    console.log("******storing failed payload in redis*********");
-    await cache.set(global.writeKey, JSON.stringify(payload));
-  }
+    // Add event to the buffer which will flush in the background
+    global.buffer.add(rudderPayload)
```

Comment:
So we stick to the 100kB limit, add the size of the payload (default is 1 byte otherwise, which isn't ok)

```suggestion
    global.buffer.add(rudderPayload, JSON.stringify(rudderPayload).length)
```